### ISSUE TITLE
More information how get Home Connect to work.

### ISSUE THREAD

**Issue by YRRK** [DavidMStraub/homeassistant-homeconnect#89]
Dear David,

I made a request to enhance the documentation in the new Home Connect official integration. I'm glad it's there. This request can be rejected :-)
Because I still can't get the Home Connect integration to work yet following the https://www.home-assistant.io/integrations/home_connect/ Probably it's my limited knowledge in this case. But now it's an official integration I want to make an attempt to really get It working. Could you help me out once and for all with answering all these noob questions?

-I remembered you needed external https acces for the beta of this integration? I couldnt get it to work back then. Is this still the case now it's an official integration that you need HTTPS? 

-If yes, does the HTTPS need to be always accesable, or just once for authorisation of the integration? So can I revoke the acces from the outside would as soon as it's done?

-I filled in: http://192.168.47.12:8123/auth/external/callback
as call back in the API as Redirect URI. Would this work, is this correct or do you mean it should use my external IP?

-Can I fake it with https://192.168...etc? 

-Can it need be made accesable from outside with the HA Duck DNS addon?

-If I don't want it to be accesable because I'm not shure what the risks are. Is the famous Nabu Casa service a safe solution? Would that work for activating this integration? They provide that safe ssl connection to their servers. 

-If so, what would be my Redirect URI : with NabuCasa as soon i've setup their services?

-My external IP is not auto-filled-in in Configuration > General > External-ip. Is that a problem?
I read this in the 0.110 release is an overrule of the auto-detect, so it' doesn't matter it's not filled in.

Hope you can get me out of this.

Raspberry pi4b
Home Assistant 0.110.1
Frontend version: 20200519.1 - latest
arch | armv7l
-- | --
chassis | embedded
dev | false
docker | true
hassio | true
host_os | HassOS 3.13
installation_type | Home Assistant
os_name | Linux
os_version | 4.19.114-v7l
python_version | 3.7.7
supervisor | 222
timezone | Europe/Amsterdam
version | 0.110.1
virtualenv | false

**Comment by DavidMStraub** [comment]
So... the deprecation of `base_url` and introduction of `internal_url` and `external_url` is a change that accidentally happened in the same release as the addition of the integration and seems to be causing some problems. Please follow this issue: https://github.com/home-assistant/core/issues/35844

Concerning HTTPS:
- You do **not need to enable external access** to your instance
- You also do not need HTTPS (this used to be a requirement for the redirect URI from the Home Connect side but has since been dropped)

The redirect URI is the URI that your browser will be redirected to after authorization which will send all the relevant info to your HA instance, so you need to make sure that, if the redirect URI is `SOMETHING/external/auth/callback`, `SOMETHING` is the URI that you can access your HA UI from **in the browser you use for the authentication**.

If you get it to work, feel free to submit a PR to the docs repo to make the documentation clearer (always easier from a user than developer perspective).

In the future, all this hassle might become obsolute when HA implements the account linking service for Home Connect, but this has not landed yet.

**Comment by YRRK** [comment]
I'm willing to think about it if I have something to add in the PR docs. But it seems I was so far off with my thinking I went back to the drawingboard and follow the manual again. Posted my details in the other github issue now.

**Comment by YRRK** [comment]
Sorry I'm re-opening this isse. And sorry that I commented off topic on: https://github.com/home-assistant/core/issues/35844

I continue from that case, where the other Home Connect from CodWild whas in the way.
I did a complete clean-install on Raspberry Pi4b, new 32GB industrial microSDcard.
Following the manual ( https://www.home-assistant.io/integrations/home_connect/ ), I get this error when I Configuration > Integrations > + Home Connect. He says need to open external website, I say oke and then I get this:

error | "unauthorized_client"
-- | --
error_description | "client or redirection URI not authorized"
state | "eyJ0etcetcalongprobablysecretcode".

## Configyaml
Added these to the config yaml (offcourse with all the secret data)...
home_connect:
  client_id: CLIENT_ID
  client_secret: CLIENT_SECRET

## Environment
Frontend version: 20200519.4 - latest
arch | armv7l
-- | --
chassis | embedded
dev | false
docker | true
hassio | true
host_os | HassOS 3.13
installation_type | Home Assistant
os_name | Linux
os_version | 4.19.114-v7l
python_version | 3.7.7
supervisor | 224
timezone | Europe/Amsterdam
version | 0.110.2
virtualenv | false

## Siemens website
This is how my Home Connect API looks on the  (the personal data is scrambled in this example, it's fake data)

Home Assistant
Created 2020-05-20 | Last Modified 2020-05-25

Access Private
   i Private access is limited to a single Home Connect user account as defined.

Home Connect user account for testing BLABLA@hotmail.com Add more
   i This Home Connect user account has been configured in your profile. You can override it for this application.

OAuth Flow: Authorization Code Grant Flow

Client ID: BLA8D3SECRETSTUFFFEC

Client Secret: (required) 75FSECRET!!!!!!!!!!D28E

Redirect URI: http://192.168.36.44:8123/external/auth/callback

One Time Token Mode Disabled

Proof Key for Code Exchange Disabled

**Comment by YRRK** [comment]
Sorry I'm re-opening this isse. And sorry that I commented off topic on: https://github.com/home-assistant/core/issues/35844

I continue from that case, where the other Home Connect from CodWild whas in the way.
I did a complete clean-install on Raspberry Pi4b, new 32GB industrial microSDcard.
Following the manual ( https://www.home-assistant.io/integrations/home_connect/ ), I get this error when I Configuration > Integrations > + Home Connect. He says need to open external website, I say oke and then I get this:

error | "unauthorized_client"
-- | --
error_description | "client or redirection URI not authorized"
state | "eyJ0etcetcalongprobablysecretcode".

## Configyaml
Added these to the config yaml (offcourse with all the secret data)...
home_connect:
  client_id: CLIENT_ID
  client_secret: CLIENT_SECRET

## Environment
Frontend version: 20200519.4 - latest
arch | armv7l
-- | --
chassis | embedded
dev | false
docker | true
hassio | true
host_os | HassOS 3.13
installation_type | Home Assistant
os_name | Linux
os_version | 4.19.114-v7l
python_version | 3.7.7
supervisor | 224
timezone | Europe/Amsterdam
version | 0.110.2
virtualenv | false

## Siemens website
This is how my Home Connect API looks on the  (the personal data is scrambled in this example, it's fake data)

Home Assistant
Created 2020-05-20 | Last Modified 2020-05-25

Access Private
   i Private access is limited to a single Home Connect user account as defined.

Home Connect user account for testing BLABLA@hotmail.com Add more
   i This Home Connect user account has been configured in your profile. You can override it for this application.

OAuth Flow: Authorization Code Grant Flow

Client ID: BLA8D3SECRETSTUFFFEC

Client Secret: (required) 75FSECRET!!!!!!!!!!D28E

Redirect URI: http://192.168.36.44:8123/external/auth/callback

One Time Token Mode Disabled

Proof Key for Code Exchange Disabled

**Comment by YRRK** [comment]
I have no clue what I just did to make it work. I'v been wiggeling around.

Added an additional callback URL: http://192.168.51.15:8123/auth/external/callback&state
Instead of http://192.168.51.15:8123/auth/external/callback

Then he asked me for my login details!
But that didn't work either. It seemed not to accept my developer ID login details...but my regular home-connect user account details, wich is different. Then it worked!

I think this is what I did. Maybe this is something to check out with a clean install and new user id different from the developer id. If you use both the same id you'd maybe nut run in to this...
But i'm just speculating now...
